packages/nimbus/src/components/text-input/text-input.dev.mdx

```diff
@@ -0,0 +1,631 @@
+        <>
+          <TextInput
+            value={value}
+            onChange={handleChange}
+            isInvalid={!isValid}
+          />
+          {!isValid && <span>Must be at least 3 characters</span>}
+        </>
+      );
+    };
+
+    render(<TestComponent />);
+
+    const input = screen.getByRole('textbox');
+    await user.type(input, 'ab');
+
+    expect(screen.getByText('Must be at least 3 characters')).toBeInTheDocument();
+
+    await user.type(input, 'c');
+
+    expect(screen.queryByText('Must be at least 3 characters')).not.toBeInTheDocument();
+  });
+});
+```
+
+### Testing leading and trailing elements
+
+Test custom elements within the input:
+
+```tsx
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { TextInput } from '@commercetools/nimbus';
+
+describe('TextInput with elements', () => {
+  it('renders leading element', () => {
+    render(
+      <TextInput
+        leadingElement={<span data-testid="icon">🔍</span>}
+        placeholder="Search"
+      />
+    );
+
+    expect(screen.getByTestId('icon')).toBeInTheDocument();
+  });
+
+  it('renders trailing element', () => {
+    render(
+      <TextInput
+        trailingElement={
+          <button data-testid="clear">Clear</button>
+        }
+        placeholder="Input"
+      />
+    );
+
+    expect(screen.getByTestId('clear')).toBeInTheDocument();
+  });
+
+  it('trailing button is interactive', async () => {
+    const user = userEvent.setup();
+    const handleClick = jest.fn();
+
+    render(
+      <TextInput
+        trailingElement={
+          <button onClick={handleClick}>Action</button>
+        }
+      />
+    );
+
+    await user.click(screen.getByText('Action'));
+
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+});
+```
+
+### Testing validation states
+
+Test different validation states:
+
+```tsx
+import { render, screen } from '@testing-library/react';
+import { TextInput } from '@commercetools/nimbus';
+
+describe('TextInput validation states', () => {
+  it('renders disabled state', () => {
+    render(<TextInput isDisabled placeholder="Disabled" />);
+
+    const input = screen.getByRole('textbox');
+    expect(input).toBeDisabled();
+  });
+
+  it('renders invalid state', () => {
+    render(<TextInput isInvalid placeholder="Invalid" />);
+
+    const input = screen.getByRole('textbox');
+    expect(input).toHaveAttribute('aria-invalid', 'true');
+  });
+
+  it('renders read-only state', () => {
+    render(<TextInput isReadOnly value="Read-only" onChange={() => {}} />);
+
+    const input = screen.getByRole('textbox');
+    expect(input).toHaveAttribute('readonly');
+  });
+
+  it('renders required state', () => {
+    render(<TextInput isRequired placeholder="Required" />);
+
+    const input = screen.getByRole('textbox');
+    expect(input).toHaveAttribute('aria-required', 'true');
+  });
+});
+```
+
+## Resources
+
+- [Storybook](link-tbd)
```

Comment:
Can we link to [here](https://nimbus-storybook.vercel.app/?path=/docs/components-textinput--docs)?